__tests__/app.test.js

```diff
@@ -93,6 +93,153 @@ describe('GET /api/articles', () => {
+      return request(app)
+        .get('/api/articles?topic=mitch')
+        .expect(200)
+        .then(({ body }) => {
+          const { articles } = body
+
+          articles.forEach((article) => {
+            expect(article.topic).toBe('mitch');
+          });
+        });
+    });
+
+    it('200: responds with an empty array if the topic exists but there are no articles with that topic', () => {
+      return request(app)
+        .get('/api/articles?topic=paper')
+        .expect(200)
+        .then(({ body }) => {
+          const { articles } = body
+          expect(articles).toBeInstanceOf(Array);
+          expect(articles).toHaveLength(0);
+        });
+    });
+
+    it('404: responds with topic not found if the topic does not exist', () => {
+      return request(app)
+        .get('/api/articles?topic=non-existent-topic')
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.message).toBe('Topic not found');
+        });
+    });
+  });
+
+  describe('?sort_by (sorted in descending order by default)', () => {
+    it('200: responds with articles sorted by article_id', () => {
+      return request(app)
+        .get('/api/articles?sort_by=article_id')
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSortedBy('article_id', { descending: true });
+        });
+    });
+
+    it('200: responds with articles sorted by title', () => {
+      return request(app)
+        .get('/api/articles?sort_by=title')
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSortedBy('title', { descending: true });
+        });
+    });
+
+    it('200: responds with articles sorted by topic', () => {
+      return request(app)
+        .get('/api/articles?sort_by=topic')
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSortedBy('topic', { descending: true });
+        });
+    });
+
+    it('200: responds with articles sorted by author', () => {
+      return request(app)
+        .get('/api/articles?sort_by=author')
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSortedBy('author', { descending: true });
+        });
+    });
+
+    it('200: responds with articles sorted by created_at', () => {
+      return request(app)
+        .get('/api/articles?sort_by=created_at')
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSortedBy('created_at', { descending: true });
+        });
+    });
+
+    it('200: responds with articles sorted by votes', () => {
+      return request(app)
+        .get('/api/articles?sort_by=votes')
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSortedBy('votes', { descending: true });
+        });
+    });
+
+    it('200: responds with articles sorted by article_img_url', () => {
+      return request(app)
```

Comment:
we don't need to test that every version of sort by is working :)

Reply:
I wanted to be thorough. :) Is just one ok? Perhaps one that is not the default?